components/scream/src/share/tests/field_tests.cpp

```diff
@@ -684,6 +685,34 @@ TEST_CASE("field_property_check", "") {
     }
   }
 
+  // Check that values are not NaN
+  SECTION("field_not_nan_check") {
+    Field<Real> f1(fid);
+    auto nan_check = std::make_shared<FieldNaNCheck<Real>>();
+    f1.add_property_check(nan_check);
+    f1.allocate_view();
+
+    // Assign  values to the field and make sure it passes our test for
+    // NaNs.
+    auto f1_view = f1.get_view();
+    auto host_view = Kokkos::create_mirror_view(f1_view);
+    for (int i = 0; i < host_view.extent_int(0); ++i) {
+      host_view(i) = i;
+    }
+    Kokkos::deep_copy(f1_view, host_view);
+    for (auto iter = f1.property_check_begin(); iter != f1.property_check_end(); iter++) {
+      REQUIRE(iter->check(f1));
+    }
+
+    // Assign a NaN value to the field, make sure it fails the check,
+    Int midpt = host_view.extent(0)/2;
+    host_view(midpt) = std::numeric_limits<Real>::quiet_NaN();
+    Kokkos::deep_copy(f1_view, host_view);
+    for (auto iter = f1.property_check_begin(); iter != f1.property_check_end(); iter++) {
```

Comment:
Minor: the loop here is pointless, since you have only one check. You could just do

```
const auto& nan_check = *f1.property_check_begin();
REQUIRE (nan_check->check(f1));
```

Reply:
per this comment, if I clean this up here, should I also clean it up for the other property check tests?  See positivity_check, monotonicity_check and within_interval_check.

Reply:
You don't have to clean up anything, this can stay the way it is...